ocfweb/docs/views/servers.py

```diff
@@ -57,81 +57,89 @@ def has_munin(self):
+        result[hostname] = d.get('value')
+    return result
+
+
+def create_hosts(lst):
+    """Accepts a list of raw ldap output, returns a dictionary of host
+    objects indexed by hostname
+    """
+    hosts = {}
+    for h in lst:
+        if not is_hidden(h):
+            description = h.get('description', [''])[0]
+            hostname = h['cn'][0]
+            hosts[hostname] = (Host(hostname, h['type'], description, ()))
+    return hosts
+
+
+@periodic(300)
 def get_hosts():
-    return [
-        Host.from_ldap(
-            hostname='hal',
-            type='hypervisor',
-            children=[
-                Host.from_ldap(hostname)
-                for hostname in (
-                    'fallingrocks',
-                    'monsoon',
-                )
-            ],
-        ),
-
-        Host.from_ldap(
-            hostname='jaws',
-            type='hypervisor',
-            children=[
-                Host.from_ldap(hostname)
-                for hostname in (
-                    'anthrax',
-                    'biohazard',
-                    'death',
-                    'dementors',
-                    'democracy',
-                    'firestorm',
-                    'flood',
-                    'lightning',
-                    'maelstrom',
-                    'pestilence',
-                    'reaper',
-                    'supernova',
-                    'thunder',
-                    'tsunami',
-                    'werewolves',
-                    'whirlwind',
-                    'whiteout',
-                    'zombies',
-                )
-            ],
-        ),
-
-        Host.from_ldap(
-            hostname='pandemic',
-            type='hypervisor',
-            children=[
-                Host.from_ldap(hostname)
-                for hostname in (
-                    'pileup',
-                )
-            ],
-        ),
-
-        Host.from_ldap(
-            hostname='riptide',
-            type='hypervisor',
-        ),
-
-        Host.from_ldap(
-            hostname='corruption',
-            type='server',
-        ),
-
-        Host('blackhole', 'network', 'Managed Cisco Catalyst 2960S-48TS-L Switch.', []),
-
-        Host('logjam', 'printer', '', []),
-        Host('pagefault', 'printer', '', []),
-        Host('papercut', 'printer', '', []),
-        Host.from_ldap('overheat', type='raspi'),
-        Host.from_ldap('tornado', type='nuc'),
-
-        *(Host.from_ldap(desktop, type='desktop')
-          for desktop in sorted(list_desktops())),
-    ]
+    servers = create_hosts(hosts_by_filter('(type=server)'))
+    desktops = create_hosts(hosts_by_filter('(type=desktop)'))
+    misc = create_hosts(hosts_by_filter('(type=printer)'))
+    hypervisors = {}
+
+    hypervisor_hostnames = query_puppet(PQL_IS_HYPERVISOR)
+    all_children = query_puppet(PQL_GET_CHILDREN)
+
+    # Add children to hypervisors
+    for h in list(servers.values()):
```

Comment:
No need to convert this to a list, Python will know to loop through the set of values.

Reply:
The list acts as a copy of the elements in servers.values() since the dictionary servers is modified during iteration.